test/model/message_list_test.dart

```diff
@@ -734,6 +734,52 @@ void main() {
       checkNotifiedOnce();
       check(model.messages.map((m) => m.id)).deepEquals(expected..add(301));
     });
+
+    test('in MentionsNarrow', () async {
+      final stream1 = eg.stream(streamId: 1, name: 'stream 1');
+      final stream2 = eg.stream(streamId: 2, name: 'stream 2');
+      await prepare(narrow: const MentionsNarrow());
+      await store.addStreams([stream1, stream2]);
+      await store.addSubscription(eg.subscription(stream1));
+      await store.addUserTopic(stream1, 'B', UserTopicVisibilityPolicy.muted);
+      await store.addSubscription(eg.subscription(stream2, isMuted: true));
+      await store.addUserTopic(stream2, 'C', UserTopicVisibilityPolicy.unmuted);
+
+      List<Message> getMessages(int startingId) => [
+        eg.streamMessage(id: startingId,
+          stream: stream1, topic: "A", flags: [MessageFlag.wildcardMentioned]),
+        eg.streamMessage(id: startingId + 1,
+          stream: stream1, topic: "B", flags: [MessageFlag.wildcardMentioned]),
+        eg.streamMessage(id: startingId + 2,
+          stream: stream2, topic: "C", flags: [MessageFlag.wildcardMentioned]),
```

Comment:
When writing a test, a question to ask is: what story does this test need to tell? What is it checking, and what scenarios should it be exercising?

Here, the story that's needed is: messages get included in the view, even messages you might think wouldn't be. Specifically that means: including even a message in a muted topic and muted channel, including even DMs, and including even messages that have flag `wildcardMentioned` but not `mentioned`.

(The other half of the story would be: messages get excluded from the view, even messages you might think would be included. But for this narrow there are no messages that get excluded, so there's nothing to say in that direction.)

So then the test data should be written to focus on that story. In particular, the messages here that are to an unmuted channel or topic aren't needed — if messages are included even when muted, surely they are when unmuted too.

(It looks like this set of data is based on the one for CombinedFeedNarrow. But there the answer is different for muted vs. unmuted, so it's important to cover both sides.)